provider/cmd/app/app.go

```diff
@@ -556,6 +557,11 @@ func initResourceManager(ctx context.Context, opts sources.Options, machineStore
 		return err
 	}
 
+	err = manager.SetVMLimit(int(opts.VMLimit))
```

Comment:
this retype can change value it isn't secure/

Reply:
type conversion is no longer needed here.